### PR TITLE
extension for DoubleClick Floodlight Tag Migration - Flight UDP

### PR DESCRIPTION
**Pull request by v-roroy**

### CODE REVIEW

File: src/library/simplified/extension/doubleclick-carriercode-flightnumber--u18.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+    b['u18'] = '';
+    if (b['carrierCodes'] != undefined && b['flightNumbers'] != undefined) {
+        var carrierCodeArray = b['carrierCodes'].split('|');
+        var flightNumArray = b['flightNumbers'].split('|');
+        var carrierFlightNum = '';
+        for (var i = 0; i < carrierCodeArray.length; i++){
+            for (var j = i; j <= i; j++){
+                carrierFlightNum = carrierFlightNum + carrierCodeArray[i] + flightNumArray[j] + '|';
```
Comment (trungnguyenexp) [inline src/library/simplified/extension/doubleclick-carriercode-flightnumber--u18.js]: Will there be a potential for indexOutOfBound for flightNumArray[j]?

File: src/library/simplified/extension/price-per-person.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ else if((utag.isHCO()|| utag.isFCO() || utag.isPCO()|| utag.isLXCO() || utag.isM
         b['pricePerPerson'] = (b['checkout.cartTotal.amount'] / b['numberOfGuests']).toFixed(2);
     }
 }
+else if (utag.isFRateDetails() && (b['entity.flight.flight.avgPriceOfAdultAndSenior.amount'] || b['entity.flight.flight.avgPriceOfChildrenAndInfants.amount']))
+{
+    b['pricePerPerson'] = b['entity.flight.flight.avgPriceOfAdultAndSenior.amount'] + "|" + b['entity.flight.flight.avgPriceOfChildrenAndInfants.amount'];
```
Comment (trungnguyenexp) [inline src/library/simplified/extension/price-per-person.js:19]: I'm not seeing entity.flight.flight.avgPriceOfAdultAndSenior.amount and entity.flight.flight.avgPriceOfChildrenAndInfants.amount on the Flight UDP page when I searched; therefore, pricePerPerson is empty in my case. I do see entity.tripDetails.flightOffer.pricePerTraveler.amount with the price. Should we also check for this field?

File: src/library/simplified/extension/trip-type.js [hunk 3]
```diff
@@ -55,7 +53,9 @@
...
         b['tripType'] = ROUND_TRIP;
     }
-    else if ((utag.isFRateDetails() || utag.isPRateDetails()) && b['entity.tripDetails.flightOffer.tripType'] != undefined){
+    else if (utag.isPRateDetails() && b['entity.tripDetails.flightOffer.tripType'] != undefined){
         b['tripType'] = getTripType(b['entity.tripDetails.flightOffer.tripType']);
     }
-})();
\ No newline at end of file
+    else if (utag.isFRateDetails() && b['entity.tripDetails.flightOffer.tripType'] != undefined){
+        b['tripType'] = b['entity.tripDetails.flightOffer.tripType']
```
Comment (trungnguyenexp) [inline src/library/simplified/extension/trip-type.js:60]: do we need to call getTripType() here?

File: src/library/simplified/extension/doubleclick-carriercode-flightnumber--u18.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+    b['u18'] = '';
```
Comment (wmoussa) [inline src/library/simplified/extension/doubleclick-carriercode-flightnumber--u18.js:6]: would be great to add an example block at the beginning of this extension so we can visualize the end results and input data

File: src/library/simplified/extension/price-per-person.js (hunk 2 above)
Comment (wmoussa) [reply src/library/simplified/extension/price-per-person.js:19]: the new fields would be available after the merge of this https://swarm-1994.sea.corp.expecn.com/changes/1745761

File: src/library/simplified/extension/trip-type.js (hunk 3 above)
Comment (v-roroy) [reply src/library/simplified/extension/trip-type.js:60]: Hi @trungnguyenexp ,
According to the doubleclick fire , the triptype (u20) looks like OW RT etc..
So I removed the getTripType()  call.
Sample fire:
doub
https://fls.doubleclick.net/activityi;src=2588797;type=nausc826;cat=nausr653;u1=Flight;u2=DEL;u3=SFO;u4=20170401|20170402;u7=3|0;u10=SQ|SQ;u11=Economy/Coach;u14=178.493|0.00;u15=1,606.44;u16=USD;u18=SQ31|SQ402**;u20=OW**;u=3d3f9656-9ea2-4f18-ae45-9f1597c80baf;ord=1;num=6312990565403.344?

**Pull request merged**